crates/libs/windows/src/core/borrowed.rs

```diff
@@ -1,50 +1,77 @@
 ///
-/// Generally a value of type `Borrowed<'a, T>` can be thought of as equivalent to `&'a T` with the only different being that 
-/// in-memory it has the exact same layout as `T`. This type is used at the FFI boundary to bridge Rust's ownership semantics
-/// and the looser semantics around ownership in the Windows API.
+/// `Borrowed`s can be thought of much like an `Option<&T>`. The reason `Borrowed` must be used instead of 
+/// `Option<&T>` is because `Borrowed` has the same in-memory layout as `T`. This is necessary
+/// for FFI calls that expect a logically borrowed type that, in-memory, looks like an owned type.
+///
+/// # Usage
+///
+/// Many functions in the `windows` crate have signatures similar to the following:
+///
+/// ```rust
+/// fn SomeFunction<'a, P: Into<Borrowed<'a, IUnknown>>>(iunknown: P);
+/// ```
+///
+/// This signature allows the parameter `iunknown` to passed any value that implements `Into<Borrowed<'a, IUnknown>>`. In other
+/// words, `SomeFunction` takes values of any type that can turn themselves into a borrowed `IUnknown`. Generally, if this
+/// is safe to do, the `windows` crate provides an implementation. Here are the typical things that can be converted into an `Borrowed<'a, T>`:
 ///
-/// # Conversion
+/// * References to a value of type `T` (i.e., `&'a T`).
+/// * Anything that can be turned into a `&'a T`.
+///   * For example, many COM interfaces have such conversions to parent interfaces. For example, if a function requires an `Borrowed<'a, IUnknown>`,
+///     you can pass a `&'a IInspectable` since `IInspectable` inherits from `IUnknown`.
+/// * `None` - because `Borrowed`s are always optional
 ///
-/// If you're looking to convert from a value of type `T` to a `Borrowed<'a, T>` there will generally be a `From` implementation 
-/// that achieves this. For example, if you want to convert an `IUnknown` to a `Borrowed<'a, IUnknown>` you simply need to call 
-/// `.into()` on that value.
+/// It's important to note that owned values **cannot** be passed as `Borrowed`s. After all, `Borrowed` types are borrowed and not owned. Just as you cannot
+/// pass a `T` into a function that expects a `&T`, you cannot pass a `T` into a function that expects a `Borrowed<'a, T>`. If you get something similar to 
+/// the following error, it's likely because you are trying to pass an owned value instead of a borrowed value:
 ///
-/// Many APIs take a params of the form `Into<Borrowed<'a, T>>`. This means you can pass anything that has an `Into` implementation 
-/// for `Borrowed<'a, T>`. Generally, if this is safe to do, the `windows` crate provides an implementation.
+/// ```
+///  the trait bound `SomeType: Into<Borrowed<'_, SomeType>>` is not satisfied
+/// ```
 #[repr(transparent)]
 pub struct Borrowed<'a, T> {
-	item: core::mem::ManuallyDrop<T>,
-    lifetime: core::marker::PhantomData<&'a ()>
+    item: Option<core::mem::ManuallyDrop<T>>,
```

Comment:
I'm not sure `Option<ManuallyDrop>` is appropriate since it won't necessarily be the same size as `T` if `T` is not a transparent pointer. `T` and `Borrowed<T>` must have the same size.

Reply:
But just storing a `ManuallyDrop` that uses `zeroed` for the `None` case should work fine but implementing `as_ref` might be tricky.

Reply:
We're now storing `T::Abi` instead.